packages/sdk-client-v3/src/middleware/create-concurrent-modification-middleware.ts

```diff
@@ -19,10 +25,14 @@ export default function createConcurrentModificationMiddleware(): Middleware {
 
         // update the resource version here
         if (version) {
-          request.body =
-            typeof request.body == 'string'
-              ? { ...JSON.parse(request.body), version }
-              : { ...request.body, version }
+          if (customLogic) {
```

Comment:
Also, let's ensure this is a function, just a little validation.
```suggestion
          if (customLogic && typeof customLogic == 'function') {
```